features/step_definitions/workflow.js

```diff
@@ -84,6 +96,28 @@ When(
     }
 );
 
+When(
+    /^a pull request is opened from "(.*)" branch$/,
+    {
+        timeout: TIMEOUT
+    },
+    function step(branch) {
+        return github
+            .createBranch(branch, this.repoOrg, this.repoName)
+            .then(() => github.createFile(branch, this.repoOrg, this.repoName))
+            .then(() => github.createPullRequest(branch, this.repoOrg, this.repoName))
+            .then(({ data }) => {
+                this.branch = branch;
+                this.pullRequestNumber = data.number;
+                this.sha = data.head.sha;
+            })
+            .catch(err => {
+                // throws an error if a PR already exists, so this is fine
+                Assert.strictEqual(err.status, 422);
```

Comment:
If the pull request already exists, isn't it set to any `this` value? It may be better to make an error to warn of unexpected situations.

Reply:
You're right! I'll try to make an error.